packages/next/src/server/route-modules/app-route/module.ts

```diff
@@ -844,7 +845,8 @@ function proxyNextRequest(request: NextRequest, workStore: WorkStore) {
         case 'toJSON':
         case 'toString':
         case 'origin': {
-          trackDynamicDataAccessed(workStore, `nextUrl.${prop}`)
+          const cacheStore = cacheAsyncStorage.getStore()
+          trackDynamicDataAccessed(workStore, cacheStore, `nextUrl.${prop}`)
```

Comment:
@gnoff This is a change but I think maybe a bug fix? Since this was closing over the outer WorkStore, this wasn't changing if it was accessed inside unstable_cache (if you somehow closed over it).